### PR TITLE
feat / allow fees in different tokens

### PR DESCRIPTION
**Pull request by aarmoa**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
In this PR we introduce the required changes for the client to accept having fees for a trade in different tokens (something that happens with Binance connector, that sometimes can charge fees in BNB and sometimes in the trade quote token). The changes include:
- Remove fee_amount and fee_token from the order complete events, since that information was not being used anywhere
- Change PNL calculation logic to convert fees in different tokens to the quote token using the RateOracle as price provider
- Modification of the start command to always start also the RateOracle, for it to regularly update prices

**Tests performed by the developer**:
Added new unit tests.
Tested manually by running a PMM strategy against Binance and verifying that PNL is calculated without issues.

**Tips for QA testing**:
Please check that all connectors can correctly process order complete events.
Verify PNL calculation.

NOTE: It would be a good idea to retest all current issues related to PNL calculation with this PR. Many of them could be solved with this changes.

[ch23042]

### CODE REVIEW

**Review by petioptrv — CHANGES_REQUESTED** [review]

**Comment by petioptrv** [comment]
I've suggested some very minor changes that will not invalidate QA tests

**Comment by RC-13** [comment]
QA update: Testing currently in progress

File: hummingbot/core/rate_oracle/rate_oracle.py [hunk 1]
```diff
@@ -186,7 +203,7 @@ async def get_prices(cls) -> Dict[str, Decimal]:
             raise NotImplementedError
 
     @classmethod
-    @async_ttl_cache(ttl=1, maxsize=1)
+    @async_ttl_cache(ttl=30, maxsize=1)
```
[the remark this answers is not in the document]
Comment (aarmoa) [reply hummingbot/core/rate_oracle/rate_oracle.py:206]: Yes, I think that is a good idea.

**Review by petioptrv — APPROVED** [review]
Changes LGTM!

**Comment by RC-13** [comment]
Update from yesterday, additional exchange tested:
Coinbase pro
Coinzoom
Crypto.com
will finish the remaining connectors today.

**Comment by RC-13** [comment]
Test update for today, finish testing the following connectors:
Digifinex
Dydx_perpetual
FTX
Gate_io
Huobi

**Review by RC-13 — APPROVED** [review]
LGTM
Test Summary:
- Test PR across different connectors except for connectors with a known issue and low volume exchange.
- Wait for both to both get a fill on bid and ask side and check the history performance.
- Check the rate of the fees paid in base asset using rate oracle 
- Manually compute the total PnL after getting the converted value of the fees paid and confirming that it matches or is close enough to the displayed total PnL
- Enable the use of a third-party asset for fee payment, for example, BNB, and wait for orders to get a filling.
- Manually convert the fees paid that are currently not in quote value and get the total fees paid.
- Manually compute for the total PnL again and confirmed that the PnL matches the output in the performance.
- Check AMM connectors as well to make sure that fees and pnl are calculated properly.

**Comment by aarmoa** [comment]
Waiting for https://github.com/hummingbot/hummingbot/pull/5120 to be merged before creating this PR in the foundation repository

**Pull request closed without merging**